controllers/author-controller.js

```diff
@@ -0,0 +1,210 @@
+var Author = require("../models/author");
+var Book = require("../models/book");
+
+var async = require("async");
+
+var { body, validationResult } = require("express-validator");
+
+// Display list of all Authors.
+exports.author_list = function (req, res) {
+  Author.find()
+    .sort([["family_name", "ascending"]])
+    .exec(function (err, list_authors) {
+      if (err) {
+        return next(err);
+      }
+      //Successful, so render
+      res.render("author-list", {
+        title: "Author List",
+        author_list: list_authors,
+      });
+    });
+};
+
+// Display detail page for a specific Author.
+exports.author_detail = function (req, res) {
```

Comment:
`const { id } = req.params;`

Please extract and store params and query fields in variables in the beginning, they may be used multiple times in the function body.

Reply:
Please make these changes through out from now for all `params`, `query` and `body` of the req.